### ISSUE TITLE
RELRO protection failed (Docker pre-release 1.10.0rc1)

### ISSUE THREAD

**Issue by rtlong** [gliderlabs/docker-alpine#133]
I'm not at all sure what's happening here, and I appreciate I'm using a pre-release of Docker so this might be premature. Nevertheless, I'd appreciate someone else giving their thoughts on what is happening here.

I can't seem to run any process in an alpine-based docker container while using the official Docker 1.10.0rc1 build:

This machine was set up using docker-machine: 
```
docker-machine create --engine-install-url 'https://test.docker.com' ...
```
```
# docker version
Client:
 Version:      1.10.0-rc1
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   677c593
 Built:        Fri Jan 15 18:42:26 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.0-rc1
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   677c593
 Built:        Fri Jan 15 18:42:26 2016
 OS/Arch:      linux/amd64
```
```
# uname -a
Linux big-docker-machine-4 3.13.0-71-generic #114-Ubuntu SMP Tue Dec 1 02:34:22 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```
```
# cat /etc/os-release
NAME="Ubuntu"
VERSION="14.04.3 LTS, Trusty Tahr"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 14.04.3 LTS"
VERSION_ID="14.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
```
```
# docker pull alpine
Using default tag: latest
latest: Pulling from library/alpine
c52e3ed763ff: Already exists
Digest: sha256:d7201bdfa765bebed9a79c385db4378f55593e50dc8d002b9713810320ad93b9
Status: Image is up to date for alpine:latest
```
Other images work fine:
```
# docker run --rm busybox true
```
(no output, as expected)

Alpine does not:
```
# docker run --rm alpine true
Error relocating /lib/ld-musl-x86_64.so.1: RELRO protection failed: Permission denied
Error relocating /bin/true: RELRO protection failed: Permission denied
```

**Comment by andyshinn** [comment]
I don't get any error output. But I certainly can't get any alpine container to run. Still researching.

**Comment by rtlong** [comment]
@andyshinn thanks for looking into it!

**Comment by andyshinn** [comment]
Some more information: it appears to be related to AppArmor in some way (possibly only related to Ubuntu). I've installed boot2docker 1.10 RC1 on my local Mac and didn't notice any issues. Disabling AppArmor on my Ubuntu 14.04 VM got it working. Still testing to see if this is a possible AppArmor profile issue that might just need to be fixed in packaging.

**Comment by andyshinn** [comment]
Strange... after unloading AppArmor, getting everything working, then re-enabling AppArmor and rebooting, I can't reproduce the problem. @rtlong can you try disabling and the re-enabling AppArmor and see if you can reproduce the issue?

**Comment by rtlong** [comment]
@andyshinn great to hear! I'll try now.

**Comment by rtlong** [comment]
I see the same, and can run at least `alpine:3.3` given all the same setup with that additional step of kicking AppArmor. To save anyone else some time, here's what I ran:

```
sudo service apparmor stop && sudo service apparmor start && reboot
```
I don't really want to run on Ubuntu anyway, so I think this is good motivation to learn a bit more about docker-machine and how one can have it bootstrap, say, a CoreOS VM.

Really appreciate your help figuring this out @andyshinn!

**Comment by andyshinn** [comment]
It is still a major platform used by a lot of people and will definitely hit people that can't fix or don't know about it pre-1.10 upgrade so thank you for reporting it. I'll do some more testing and see if I can pinpoint the profile difference.

**Comment by andyshinn** [comment]
OK, it looks like maybe we just uncovered this earlier than most. It looks like it is related to https://github.com/docker/docker/issues/19707.

**Comment by andyshinn** [comment]
Ah, no it was uncovered a while ago. Looks like the original main thread for the underlying broken AppArmor profile is https://github.com/docker/docker/issues/18948. The fix is at https://github.com/docker/docker/pull/19423 and should be in the 1.10 RC2 release (which sounds like it is right around the corner).

**Comment by andyshinn** [comment]
This was fixed before the 1.10 release. They removed the broken AppArmor stuff.
